File: src/InstaChatLinux/AboutDialog.cs
using Gtk;

namespace InstaChat;

/// <summary>Small non-resizable About dialog, mirroring the Windows one.</summary>
public sealed class AboutDialog : Window
{
    public AboutDialog(Window parent)
    {
        Title = "About InstaChat";
        Resizable = false;
        Modal = true;
        SetTransientFor(parent);
        SetDefaultSize(440, 0);

        var box = Gtk.Box.New(Orientation.Vertical, 10);
        box.MarginTop = 24;
        box.MarginBottom = 24;
        box.MarginStart = 24;
        box.MarginEnd = 24;

        var title = new Label { Wrap = true, Halign = Align.Start };
        title.SetMarkup("<span size='x-large' weight='bold'>InstaChat</span>");

        var version = new Label { Wrap = true, Halign = Align.Start };
        version.SetLabel($"Version {MainWindow.VersionLabel}");
        version.AddCssClass("dim-label");

        var description = new Label { Wrap = true, Halign = Align.Start };
        description.SetLabel("Chat-only Instagram. Direct messages without Reels, Explore, Feed or Stories.");

        var privacyHeader = new Label { Wrap = true, Halign = Align.Start };
        privacyHeader.SetMarkup("<span weight='bold'>Privacy</span>");

        var privacy = new Label { Wrap = true, Halign = Align.Start };
        privacy.SetLabel("• Your Instagram password is never seen or stored — you sign in on Instagram's own secure page.\n"
                       + "• Session cookies are kept in a dedicated local browser profile under ~/.local/share/InstaChat.\n"
                       + "• The app only loads the Direct Messaging experience; navigation to any other part of Instagram is blocked.\n"
                       + "• No analytics, no advertising SDKs, no data collection.");

        var close = Gtk.Button.NewWithLabel("Close");
        close.Halign = Align.End;
        close.OnClicked += (_, _) => Close();

        box.Append(title);
        box.Append(version);
        box.Append(description);
        box.Append(privacyHeader);
        box.Append(privacy);
        box.Append(close);
        SetChild(box);
    }
}

File: android/app/src/main/java/com/instachat/app/ui/AboutDialog.kt
package com.instachat.app.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.Dialog

@Composable
fun AboutDialog(onDismiss: () -> Unit) {
    Dialog(onDismissRequest = onDismiss) {
        Surface(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            shape = RoundedCornerShape(12.dp),
            color = Color(0xFFFAFAFA)
        ) {
            Column(
                modifier = Modifier.padding(24.dp)
            ) {
                Text(
                    text = "InstaChat",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.Black
                )
                Text(
                    text = "Version 1.0.0",
                    fontSize = 14.sp,
                    color = Color(0xFF666666),
                    modifier = Modifier.padding(top = 2.dp, bottom = 14.dp)
                )
                Text(
                    text = "Chat-only Instagram. Direct messages without Reels, Explore, Feed or Stories.",
                    fontSize = 14.sp,
                    color = Color.Black,
                    modifier = Modifier.padding(bottom = 16.dp)
                )
                Text(
                    text = "Privacy",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.Black,
                    modifier = Modifier.padding(bottom = 6.dp)
                )
                Text(
                    text = "\u2022 Your Instagram password is never seen or stored \u2014 you sign in on Instagram\u2019s own secure page.\n" +
                           "\u2022 Session cookies are kept in an encrypted local browser profile.\n" +
                           "\u2022 The app only loads the Direct Messaging experience; navigation to any other part of Instagram is blocked.\n" +
                           "\u2022 No analytics, no advertising SDKs, no data collection.",
                    fontSize = 13.sp,
                    color = Color(0xFF444444),
                    lineHeight = 18.sp
                )
                Spacer(modifier = Modifier.height(20.dp))
                Button(
                    onClick = onDismiss,
                    modifier = Modifier.align(Alignment.End),
                    colors = ButtonDefaults.buttonColors(
                        containerColor = Color(0xFF1976D2)
                    )
                ) {
                    Text("Close", color = Color.White)
                }
            }
        }
    }
}
